### DISCUSSION TITLE
Mixpanel: Tracking wallet project with better pattern

### DISCUSSION THREAD

**Discussion by icerove** [near/near-wallet#1517] [Ideas]
Current Tracking Pattern:
1. Tracking buttons and links with `track` function with track name and property if necessary
2. Tracking function process with `withTracking` function 
3. Tracking user profile in different places since cannot get them from one place, the pattern is with `identity` and `people` function.

Pain point:
1. It is really hard to maintain if new feature comes
2. If something breaks tracking function, cannot find out at once.

Old Tracking Pattern:
create middleware function in the middleware/index.js and then track all actions. every action will trigger double for work and cannot get the button and links click totally. Cannot create fully user flow from actions.

**Comment by marcinbodnar** [comment]
Could you point out in some king of categories what we want to track - for example, do we want to track every button, every route, only some redux actions (which one in general), etc.

**Reply by icerove** [reply]
have a look at this, this is the plan for tracking. https://docs.google.com/document/d/1OOHKZSISB0vJmlGuuzCKl7JLgph4ViK0Iv1aX3OVkUc/edit#heading=h.5a6kcggmxc8q
